### PR TITLE
feat(storage-proofs): test Feistel permutation for collisions

### PR DESCRIPTION
**Pull request by schomatis**

### CODE REVIEW

File: storage-proofs/src/zigzag_graph.rs [hunk 1]
```diff
@@ -542,4 +542,26 @@ mod tests {
...
+        let mut shuffled: HashSet<u64> = HashSet::with_capacity((n * d) as usize);
+
+        let feistel_keys = &[1, 2, 3, 4];
+        let feistel_precomputed = feistel::precompute((n * d) as feistel::Index);
+
+        for i in 0..n {
+            for k in 0..d {
+                let permuted =
+                    feistel::permute(n * d, i * d + k, feistel_keys, feistel_precomputed);
+                assert!(shuffled.insert(permuted));
```
Comment (dignifiedquire) [inline storage-proofs/src/zigzag_graph.rs]: can you add an explanation why this assert has to hold please?

File: storage-proofs/src/zigzag_graph.rs [hunk 2]
```diff
@@ -542,4 +542,26 @@ mod tests {
...
+            }
+        }
+        assert_eq!(shuffled.len(), (n * d) as usize);
```
Comment (dignifiedquire) [inline storage-proofs/src/zigzag_graph.rs]: same here

File: storage-proofs/src/zigzag_graph.rs (hunk 1 above)
Comment (schomatis) [reply storage-proofs/src/zigzag_graph.rs]: indeed i should because that's exactly what i was trying to figure out with these tests

Comment (schomatis) [reply storage-proofs/src/zigzag_graph.rs]: done

**Pull request merged**